jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebInfConfiguration.java

```diff
@@ -714,9 +715,13 @@ public static String getCanonicalNameForWebAppTmpDir (WebAppContext context)
 
         List<Resource> jarResources = new ArrayList<Resource>();
         Resource web_inf_lib = web_inf.addPath("/lib");
-        if (web_inf_lib.exists() && web_inf_lib.isDirectory())
+        if (web_inf_lib.exists() && web_inf_lib.isDirectory()) // TODO perhaps redundant given null check from list()?
         {
             String[] files=web_inf_lib.list();
+            if (files != null)
```

Comment:
Method could be flattened to returning null after each negative check for existence of JAR resources, but I elected to keep the diff shorter.